src/main/java/com/vmware/vipclient/i18n/messages/api/opt/local/LocalProductOpt.java

```diff
@@ -35,27 +35,28 @@ public LocalProductOpt(BaseDTO dto) {
     public void getSupportedLocales(MessageCacheItem cacheItem) {
         List<String> supportedLocales = new ArrayList<String>();
         try {
-
             Path path = Paths.get(VIPCfg.getInstance().getOfflineResourcesBaseUrl());
 
             URI uri = Thread.currentThread().getContextClassLoader().
                     getResource(path.toString()).toURI();
 
             if (uri.getScheme().equals("jar")) {
-                try (FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.<String, Object>emptyMap())) {
-                    path = fileSystem.getPath(path.toString());
-                    getSupportedLocales(path, supportedLocales);
+                synchronized (LocalFileSystem.getInstance()) {
```

Comment:
What's the purpose of this synchronized block?

Reply:
Pasting my reply to Xiaochao above:

_"FileSystems.newFilesystem is not thread safe so we want only 1 thread to be able to do it at a time. FileSystemAlreadyExistsException exception is thrown if another thread is simultaneously accessing the file system, making the current thread fail. We do not want the current thread to fail. Instead, we want it to wait for the other thread to finish so it could access the file system properly. To test this, have multiple threads try to access the file system (e.g. 100 calls that will get the list of supported locales at the same time.)"_

Reply:
I have changed this and moved the handling of thread safety in ProductService. Please check synchronized blocks in ProductService.

Reply:
Let's narrow down the scope to deliver this faster.  Since this fix is not related to the Atlas bug, I have removed it from this PR and created this issue:  https://github.com/vmware/singleton/issues/910.